src/plugins/mer/meremulatordevicedconfoptionswidget.cpp

```diff
@@ -0,0 +1,213 @@
+    ui->iconSubdirLineEdit->setText(value.toString());
+
+    value = getTextEditValue(DEFAULT_ALLOWED_ORIENTATIONS);
+    const auto orientation = static_cast<Orientation>(value.toInt());
+
+    ui->orientationPortraitCheckBox->setChecked(orientation & Orientation::Portrait);
+    ui->orientationPortraitInvertedCheckBox->setChecked(orientation & Orientation::PortraitInverted);
+    ui->orientationLandscapeCheckBox->setChecked(orientation & Orientation::Landscape);
+    ui->orientationLandscapeInvertedCheckBox->setChecked(orientation & Orientation::LandscapeInverted);
+}
+
+QVariant MerEmulatorDeviceDconfOptionsWidget::getTextEditValue(const QString &name) const
+{
+    const QString text = ui->optionsTextEdit->toPlainText();
+    const QRegularExpression re(QString("%1\\s*=\\s*(.*)").arg(name));
+    const QRegularExpressionMatch match = re.match(text);
+
+    if (match.hasMatch()) {
+        QString str = match.captured(1);
+        if (!str.isEmpty()) {
+            if (str.startsWith("'") && str.endsWith("'"))
+                str = str.mid(1, str.size() - 2);
+            return QVariant(str);
+        }
+    }
+
+    return QVariant();
+}
+
+void MerEmulatorDeviceDconfOptionsWidget::updateTextEdit(const QString &name, const QVariant &value)
+{
+    QString text = ui->optionsTextEdit->toPlainText();
+    const QRegularExpression re(QString("%1\\s*=\\s*(.*)\\n").arg(name));
+    const QRegularExpressionMatch match = re.match(text);
+
+    if (match.hasMatch()) {
+        if (match.captured(1).isEmpty() || value == 0 || value == "''") {
+            text.remove(match.capturedStart(), match.capturedLength()); // remove entire matched string
+            if (text.isEmpty())
+                text = SCHEMA + "\n";
+        } else {
+            text.replace(match.capturedStart(1), match.capturedLength(1), value.toString());
+        }
+    } else {
+        if (value != 0 && value != "''")
+            text.append(QString("%1=%2\n").arg(name).arg(value.toString()));
+    }
+
+    ui->optionsTextEdit->setPlainText(text);
+}
+
+void MerEmulatorDeviceDconfOptionsWidget::onSimpleOptionsChanged()
+{
+    updateTextEdit(THEME_PIXEL_RATIO, ui->pixelRatioSpinBox->value());
+    updateTextEdit(WEBVIEW_CUSTOMLAYOUTWIDTH_SCALINGFACTOR, ui->scalingFactorSpinBox->value());
+    updateTextEdit(THEME_ICON_SUBDIR, QString("'%1'").arg(ui->iconSubdirLineEdit->text()));
+
+    const auto orientation = Orientation((ui->orientationPortraitCheckBox->isChecked() ? Orientation::Portrait : Orientation::None)
+                                         | (ui->orientationPortraitInvertedCheckBox->isChecked() ? Orientation::PortraitInverted : Orientation::None)
+                                         | (ui->orientationLandscapeCheckBox->isChecked() ? Orientation::Landscape : Orientation::None)
+                                         | (ui->orientationLandscapeInvertedCheckBox->isChecked() ? Orientation::LandscapeInverted : Orientation::None));
```

Comment:
You can use QFlags::setFlag and any newer APIs available in Qt 5.9.7 as we discussed before.

Reply:
Changed to QFlags.